homeassistant/components/workday/binary_sensor.py

```diff
@@ -113,17 +113,24 @@ async def async_setup_entry(
     """Set up the Workday sensor."""
     add_holidays: list[DateLike] = entry.options[CONF_ADD_HOLIDAYS]
     remove_holidays: list[str] = entry.options[CONF_REMOVE_HOLIDAYS]
-    country: str = entry.options[CONF_COUNTRY]
+    country: str | None = entry.options.get(CONF_COUNTRY)
```

Comment:
```suggestion
    country: str = entry.options[CONF_COUNTRY]
```
This key should always be there I think even with a `None` value

Reply:
Here I'm following the pattern from the `province` value, but it makes sense to change from `get()` to `[]` if it will always be set. However, I think that the `| None` type hint should remain, since `None` _is_ an acceptable value for the `country` with this changeset.

Reply:
`province` key is missing if the province is not set. For example not all countries have province/state/etc.
`country` key should never be missing. It should be set to `None` if selected as such